docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+
+Versioned workflows with the Temporal Worker Controller solve these problems by:
+
+- ✅ **Safe Deployments**: New versions run alongside old ones, ensuring running workflows complete successfully
+- ✅ **Automated Rollouts**: Progressive rollout strategies reduce risk of new deployments
+- ✅ **Easy Rollbacks**: Can instantly route new workflows back to previous versions
+- ✅ **Workflow Continuity**: Running workflows can continue on their original version until completion
+
+## Prerequisites
+
+Before starting the migration, ensure you have:
+
+- ✅ **Unversioned Temporal workers**: Currently running workers without Worker Versioning
+- ✅ **Kubernetes cluster**: Running Kubernetes 1.19+ with CustomResourceDefinition support
+- ✅ **Basic worker configuration**: Workers connect to Temporal with namespace and task queue configuration
+- ✅ **Administrative access**: Ability to install Custom Resource Definitions and controllers
+- ✅ **Deployment pipeline**: Existing CI/CD system that can be updated to use new deployment method
+
+### Current Environment Variables
+
+Your workers are likely configured with basic environment variables like:
+
+```bash
+TEMPORAL_HOST_PORT=your-namespace.tmprl.cloud:7233
+TEMPORAL_NAMESPACE=your-temporal-namespace
+# No TEMPORAL_DEPLOYMENT_NAME or WORKER_BUILD_ID yet
+```
+
+The controller will automatically add the versioning-related environment variables during migration.
+
+## Understanding the Differences
+
+### Before: Unversioned Workers
+
+```yaml
+# Single deployment, all workers run the same code
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.3
+        env:
+        - name: TEMPORAL_HOST_PORT
+          value: "production.tmprl.cloud:7233"
+        - name: TEMPORAL_NAMESPACE
+          value: "production"
+        # No versioning environment variables
+```
+
+**Current Deployment Process:**
+1. Build new worker image
+2. Update Deployment with new image
+3. Kubernetes rolls out new pods, terminating old ones
+4. **Risk**: Running workflows may fail if code changes break compatibility
+
+### After: Versioned Workers with Controller
+
+```yaml
+# Single CRD manages multiple versions automatically
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: production
+  cutover:
+    strategy: Progressive  # Gradual rollout of new versions
+    steps:
+      - rampPercentage: 10
+        pauseDuration: 5m
+      - rampPercentage: 50
+        pauseDuration: 10m
+  sunset:
+    scaledownDelay: 1h
+    deleteDelay: 24h
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.4  # Update this to deploy new versions
```

Comment:
maybe this comment or another comment could incorporate some thing like "Updating any part of the pod spec will generate a new Target Build ID for the worker deployment"

Reply:
I just want people to know that too

Reply:
Reworded, see what you think.